src/main/scala/it/pagopa/interop/purposeprocess/service/impl/PurposeManagementServiceImpl.scala

```diff
@@ -131,6 +131,17 @@ final case class PurposeManagementServiceImpl(invoker: PurposeManagementInvoker,
     )
   }
 
+  def rejectPurposeVersion(purposeId: UUID, versionId: UUID, payload: RejectPurposeVersionPayload)(implicit
+    contexts: Seq[(String, String)]
+  ): Future[Unit] = withHeaders { (bearerToken, correlationId) =>
+    val request =
+      api.rejectPurposeVersion(xCorrelationId = correlationId, purposeId, versionId, payload)(BearerToken(bearerToken))
+    invoker.invoke(
+      request,
+      s"Rejecting Version $versionId of Purpose $purposeId with reason ${payload.rejectionReason}"
```

Comment:
As rule of thumb we avoid logging free input text
```suggestion
      s"Rejecting Version $versionId of Purpose $purposeId"
```